### ISSUE TITLE
idevicedebug very slow logging

### ISSUE THREAD

**Issue by nchapp** [libimobiledevice/libimobiledevice#210]
This may already be a known issue, but I have found that running and debugging an app using idevicedebug is not practical because of how slowly it runs. Currently if I run an app using idevicedebug the logs I see are are always extremely delayed from what is actually occurring on the device.

**Comment by AntonShpk** [comment]
look at 
tools/idevicedebug.c 
line 467: sleep(1)

**Comment by dolphineye** [comment]
Hi, we have the same issue here. Any update on this topic? Regards!

**Comment by oddgames-david** [comment]
Also experiencing this

**Comment by nikias** [comment]
@AntonShpk the sleep is in another thread, this just waits until devices appear/disappear via the registered callback handler, or ctrl+c is pressed.

I can't confirm this on my end. The data is printed as it arrives, additionally flushed to stdout whenever a linebreak is received. There's no guarantee the device sends the log output without any delays, but for me it is and has always been pretty instant.

What platform are you guys on?

**Comment by oddgames-david** [comment]
@nikias Windows 10, tried with iPhone 5 iOS 10.3.3 and iPhone 6S iOS 11.4.1

Running the app from Xcode gives me instant logs but ilibimobiledevice is very delayed, can take a few minutes to get past the start of my app which should print out in a 1-3 seconds.

Heres what it looks like (there is some legitimate delay between a couple of the lines but it's nowhere near whats' shown).

![2019-02-27_15-24-26](https://user-images.githubusercontent.com/1932174/53475611-574d1980-3aab-11e9-8843-97dafeee9f0c.gif)

**Comment by nikias** [comment]
@oddgames-david what about idevicesyslog on macOS? Windows has always been strange...

**Comment by oddgames-david** [comment]
@nikias idevicesyslog performs perfectly on Windows but it doesn't give me any useful logs from my Unity app, it's all just input, wifi, battery levels etc.

I've not tried any of the tools on osx, just output from xcode.

**Comment by nikias** [comment]
Ohh I totally misread idevicedebug for idevicesyslog. Now it makes sense.

**Comment by nikias** [comment]
Change the sleep(1) to something like usleep(10000), or Sleep(10) on windows.

**Comment by oddgames-david** [comment]
Ok, I guess it's time to leave the safety of my C# skillset and experience some C 👍  I think I see the problem with x*1000

**Comment by nikias** [comment]
The problem is not the x*1000, but the sleep in general being too long with 1 second

**Comment by oddgames-david** [comment]
@nikias It only took my 7 hours but I finally got something building with that sleep changed to a lower value and it definitely presents the characters that I was seeing previously faster but it still pauses in the same spots half way between messages.

**Comment by oddgames-david** [comment]
idevicedebug has the same speed on osx (installed latest via homebrew)

![2019-02-28_08-19-58](https://user-images.githubusercontent.com/1932174/53532399-bce3e900-3b31-11e9-9668-7a6f0baa5bc5.gif)

**Comment by nikias** [comment]
Yes I know. As I said I confused it for idevicesyslog. I think it needs another thread. Will try to look at it tomorrow...

**Comment by oddgames-david** [comment]
Any progress on this one?

**Comment by Claytonious** [comment]
Is there any news on this problem?

**Comment by anton-kl** [comment]
One workaround is to run the app with `OS_ACTIVITY_DT_MODE=enable` like this:
`idevicedebug -e OS_ACTIVITY_DT_MODE=enable run com.my.app`
This will ensure NSLog, CFLog and os_log messages get mirrored to stderr (see https://github.com/ios-control/ios-deploy/blob/master/src/scripts/lldb.py#L55)
Then you can use `idevicesyslog` to get logs from device, which will include logs from your app - in real time.
